4788/src/main/cpp/Shooter.cpp

```diff
@@ -1,82 +1,98 @@
 
 using namespace wml;
 using namespace wml::controllers;
 
-Shooter::Shooter(RobotMap::ShooterSystem &shooterSystem, SmartControllerGroup &contGroup) : _shooterSystem(shooterSystem), _contGroup(contGroup) {}
+Shooter::Shooter(RobotMap::ShooterSystem &shooterSystem, SmartControllerGroup &contGroup) : _shooterSystem(shooterSystem), _contGroup(contGroup) {
+  ControlMap::shooterPID = false;
+}
 
-void Shooter::teleopOnUpdate(double dt) {
-  // TODO @Anna decide which case to switch to
+void Shooter::setFlywheel(double power) {
+  _teleopShooter = TeleopShooter::kManual;
+  setFlyWheelPower = power;
+}
 
-  switch (_teleopShooter) {
-    case TeleopShooter::kAuto:
-      //left bumper for close shot, right bumper for far shot, POV button 
-      // if (_contGroup.Get(ControlMap::shortShoot)) {
-      //   speed(8, dt);
-      // }
-      break;
-    case TeleopShooter::kIdle:
-
-      break;
-    case TeleopShooter::kManual:
-
-      manualControl(dt);
-
-      break;
-    case TeleopShooter::kTesting:
-      testing(dt);
-      break;
-    default:
-      break;
-  }
+void Shooter::setIndex(double power) {
+  setIndexPower = power;
+}
+
+void Shooter::setAuto(double goal, double dt) {
+  pidShooterSpeed = speed(flyWheelGoal, dt);
+  _teleopShooter = TeleopShooter::kAuto;
+  flyWheelGoal = goal;
+  //call the PID loop
 }
 
-//TODO @Anna figure out PID algorithm stuff
-/**
-  * Needs to be a closed loop PID system 
-  * 
-  */
-double Shooter::speed(double metersPerSecond, double dt) {
+void Shooter::updateShooter(double dt) {
 
-  // double input = _shooterSystem._flyWheel.encoder->GetAngularVelocity();
+  switch (_teleopShooter) {
+  case TeleopShooter::kManual:
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(setFlyWheelPower * 12);
 
-  // ControlMap::error = ControlMap::goal - input;
-  // ControlMap::derror = (ControlMap::error - ControlMap::previousError) / dt;
-  // ControlMap::sum = ControlMap::sum + ControlMap::error * dt;
+    _shooterSystem.indexWheel.Set(setIndexPower);
+    std::cout << "index speed" << setIndexPower << std::endl;
 
-  // ControlMap::ouput = ControlMap::kp * ControlMap::error + ControlMap::ki * ControlMap::sum + ControlMap::kd * ControlMap::derror;
-  // ControlMap::previousError = ControlMap::error;
 
-  // return ControlMap::output;
-}
+    break;
+
+  case TeleopShooter::kIdle:
+
+    _shooterSystem.indexWheel.Set(setIndexPower);
 
 
-/**
-  * Left trigger controls the shooter manually
-  */
-void Shooter::manualControl(double dt) {
-  shooterManualSpeed = fabs(_contGroup.Get(ControlMap::shooterManualSpin)) > ControlMap::triggerDeadzone ? _contGroup.Get(ControlMap::shooterManualSpin) : 0;
+    break;
 
-  _shooterSystem.shooterGearbox.transmission->SetVoltage(shooterManualSpeed);
+  case TeleopShooter::kTesting:
+
+  
+
+    break;
+
+  case TeleopShooter::kAuto:
+    _shooterSystem.indexWheel.Set(setIndexPower);
+
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(pidShooterSpeed);
```

Comment:
You should calculate the PID value (`speed()`) each time right before this line.